### PR TITLE
DS-3179: Add a check to give a warning and sensible output when a nam…

### PR DESCRIPTION
**Pull request by chschan**
…ed palette is used with unnamed data.

### CODE REVIEW

File: tests/testthat/test-usetemplate.R [hunk 1]
```diff
@@ -95,6 +100,12 @@ test_that("Scatter plot brand names",
     res <- GetVectorOfColors(tmp.template, scatter.dat, NULL, "Scatter", 4,
                 palette = "Default or template settings")
     expect_equal(res, checkColors(col.vec))
+
```
Comment (JustinCCYap) [inline tests/testthat/test-usetemplate.R:103]: Can there be a test that verifies that the correct colours are returned by `GetVectorOfColors` in this case?

Comment (chschan) [reply tests/testthat/test-usetemplate.R:103]: Added a check for the return values in ea08b0d

File: tests/testthat/test-usetemplate.R [hunk 2]
```diff
@@ -95,12 +100,26 @@ test_that("Scatter plot brand names",
...
+                palette = "Default or template settings"),
+                "The template contains named colors but the data series is unnamed. Try selecting 'Data Series' > 'Multiple colors within a single series'")
+    expect_equal(res, c(Other = "grey"))
       
     # Using Custom palette instead of template 
     res <- GetVectorOfColors(NULL, scatter.dat, NULL, "Scatter", 4, 
         palette = "Custom palette", palette.custom.palette = rev(col.vec)[1:3])
     expect_equal(res, c(Coke = "#CCCCCC", Pepsi = "#0000FF", Fanta = "#FFA500", 
                 Sprite = "#00FF00"))
+    res <- GetVectorOfColors(NULL, 1:10, NULL, "Scatter", 4,
```
Comment (JustinCCYap) [inline tests/testthat/test-usetemplate.R:124]: Where you intending to call `expect_equal` with this `res`?

Comment (chschan) [reply tests/testthat/test-usetemplate.R:124]: Added check for `res` in 26b85a6

**Pull request merged**